Sources/InteroperabilityTests/TestService.swift

```diff
@@ -51,6 +51,18 @@ public struct TestService: Grpc_Testing_TestService.ServiceProtocol {
   public func unaryCall(
     request: ServerRequest.Single<Grpc_Testing_TestService.Method.UnaryCall.Input>
   ) async throws -> ServerResponse.Single<Grpc_Testing_TestService.Method.UnaryCall.Output> {
+    // We can't validate messages at the wire-encoding layer (i.e. where the compression byte is
+    // set), so we have to check via the encoding header. Note that it is possible for the header
+    // to be set and for the message to not be compressed.
+    if request.message.expectCompressed.value,
+      !request.metadata.contains(where: { $0.key == "grpc-encoding" })
```

Comment:
We need to be careful here: "grpc-encoding" can have the value "identity" meaning not compressed.

If `expectCompressed` is `true` then I think it's probably sufficient to get all string values for "grpc-encoding" and then check that "identity" isn't the only value.